packages/state-transition/src/block/processAttestationsAltair.ts

```diff
@@ -76,19 +80,23 @@ export function processAttestationsAltair(
     // In epoch processing, this participation info is used to calculate balance updates
     let totalBalanceIncrementsWithWeight = 0;
     const validators = state.validators;
-    for (const index of attestingIndices) {
-      const flags = epochParticipation.get(index);
+    for (const validatorIndex of attestingIndices) {
+      const flags = epochParticipation.get(validatorIndex);
 
       // For normal block, > 90% of attestations belong to current epoch
       // At epoch boundary, 100% of attestations belong to previous epoch
       // so we want to update the participation flag tree in batch
 
       // Note ParticipationFlags type uses option {setBitwiseOR: true}, .set() does a |= operation
-      epochParticipation.set(index, flagsAttestation);
+      epochParticipation.set(validatorIndex, flagsAttestation);
       // epochParticipation.setStatus(index, newStatus);
 
       // Returns flags that are NOT set before (~ bitwise NOT) AND are set after
       const flagsNewSet = ~flags & flagsAttestation;
+      if (flagsNewSet !== 0) {
```

Comment:
A side note I was thinking if a slashed validator should be counted in `newSeenAttesters` and `newSeenAttestersEffectiveBalance`. 

The answer should be yes, as proposer gets reward by including votes from slashed validators as well. It's just that their votes will be ignored during fork choice.